File: src/main/year2015/day02/Day02.java
package year2015.day02;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.stream.Collectors;

public class Day02 {

    private List<String> input;

    public Day02(String text) {
        String day = "02"; //day
        String year = "2015"; //year
        try {
            String filePath = "src/resources/" + year + "/" + day + "/" + text;
            input = Files.readAllLines(Paths.get(filePath));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public int part1() {
        int paper = 0;
        for (String s : input) {
            int length = Integer.parseInt(s.split("x")[0]);
            int width = Integer.parseInt(s.split("x")[1]);
            int height = Integer.parseInt(s.split("x")[2]);
            int area = 2 * length * width + 2 * width * height + 2 * height * length;
            int smallestSide = Math.min(Math.min(length * width, width * height), height * length);
            paper += area + smallestSide;
        }
        return paper;
    }


    public int part2() {
        int ribbon = 0;
        for (String s : input) {
            int length = Integer.parseInt(s.split("x")[0]);
            int width = Integer.parseInt(s.split("x")[1]);
            int height = Integer.parseInt(s.split("x")[2]);
            int smallestPerimeter = Math.min(Math.min(2 * length + 2 * width, 2 * width + 2 * height), 2 * height + 2 * length);
            int volume = length * width * height;
            ribbon += smallestPerimeter + volume;
        }
        return ribbon;
    }

    public static void main(String[] args) {
        Day02 day02 = new Day02("input.txt");
        System.out.println("Part 1: " + day02.part1());
        System.out.println("Part 2: " + day02.part2());
    }
}

File: src/main/year2019/day02/Day02.kt
package main.year2019.day02

import java.io.File

class Day02(text : String = "input.txt") {

    private val day = "02"
    private val year = "2019"

    private val input = File("src/resources/$year/$day/$text").readText().split(",").map { it.toInt() }.toIntArray()


    private fun calculate(input: IntArray): Int {
        for (i in input.indices step 4) {
            if (input[i] == 99) break
            val a = input[input[i + 1]]
            val b = input[input[i + 2]]
            val c = input[i + 3]
            input[c] = if (input[i] == 1) a + b else a * b
        }
        return input[0]
    }

    fun part1(): Int {
        val input = input.copyOf()
        if (input.size > 12) {
            input[1] = 12
            input[2] = 2
        }
        return calculate(input)
    }

    fun part2(): Int {
        for (i in 0..99) {
            for (j in 0..99) {
                val input = input.copyOf()
                input[1] = i
                input[2] = j
                if (calculate(input) == 19690720) return 100 * i + j
            }
        }
        return 0
    }
}

